src/results_analyze/analyzer_plugin.py

```diff
@@ -0,0 +1,133 @@
+    type information for each match errors.
+    """
+
+    resource_attributes = {
+        "license_detection_analysis": attr.ib(default=attr.Factory(list))
+    }
+
+    sort_order = 80
+
+    options = [
+        PluggableCommandLineOption(
+            ("--analyze-results",),
+            is_flag=True,
+            default=False,
+            help='Add the "license_detection_analysis" list which has the'
+            "license detection error type information for each match errors",
+            help_group=POST_SCAN_GROUP,
+        ),
+    ]
+
+    def is_enabled(self, analyze_results, **kwargs):
+        return analyze_results
+
+    def process_codebase(self, codebase, **kwargs):
+        for resource in codebase.walk():
+            if not resource.is_file:
+                continue
+
+            try:
+                # Will fail if missing attributes
+                if validate_resource(resource):
+                    resource.license_detection_analysis = analyze_resource(resource)
+                    codebase.save_resource(resource)
+            except NotAnalyzableResourceException as e:
+                msg = str(e)
+                codebase.errors.append(msg)
+                break
+
+
+def validate_resource(resource):
+    """
+    Return True if resource has all the data required for the analysis.
+    Return False if the resource does not have detected licenses.
+    Raise an exception if any of the essential attributes are missing from the resource.
+    """
+    has_licenses = hasattr(resource, "licenses")
+    licenses = getattr(resource, "licenses", [])
+    if has_licenses and not licenses:
+        return False
+
+    has_license_text = hasattr(resource, "is_license_text")
+    has_legal = hasattr(resource, "is_legal")
+    has_matched_text = all(
+        "matched_text" in license_match for license_match in licenses
+    )
+
+    if has_licenses and has_license_text and has_matched_text and has_legal:
+        return True
+
+    raise NotAnalyzableResourceException(
+        f"{resource.path} cannot be analyzed for license scan errors, "
+        f"required attributes are: is_license_text, is_legal, license.matched_text. "
+        f"Rerun scan with these options: --license --license-text --is-license-text --classify --info"
+    )
+
+
+def analyze_resource(resource):
+    """
+    Analyzes license scan attributes for a resource and classifies license scan issues.
+
+    :param resource: object
+        An object of the commoncode.Resource class, having all resource level scan-data.
+    :return: dict
+        Resource attribute containing license scan analysis result for the resource.
+    """
+    has_attributes_for_analysis = validate_resource(resource)
+
+    if not has_attributes_for_analysis:
+        return []
+
+    licenses = getattr(resource, "licenses")
+    is_license_text = getattr(resource, "is_license_text", False)
+    is_legal = getattr(resource, "is_legal", False)
+
+    matched_licences = licenses
```

Comment:
You could assign the value from the resource directly to `matched_licences`, no need to assign it to `licenses` first.

Reply:
Yes! that's a silly mistake, doing that.